rails/react-components/src/library/components/permission-forms-v2/students-tab/classes-table.tsx

```diff
@@ -8,14 +12,59 @@ interface IProps {
   currentSelectedProject: CurrentSelectedProject;
 }
 
-export const ClassesTable = ({ teacherId }: IProps) => {
+export const ClassesTable = ({ teacherId, currentSelectedProject }: IProps) => {
+  const { data: classesData, isLoading } = useFetch<IClassBasicInfo[]>(Portal.API_V1.teacherClasses(teacherId), []);
+  const [selectedClassId, setSelectedClassId] = useState<string | null>(null);
+
+  if (isLoading) {
+    return (<div>Loading...</div>);
+  }
+  if (!classesData.length) {
+    return (<div>No classes found</div>);
+  }
+
+  const handleViewStudentsClick = (classId: string) => {
+    setSelectedClassId((prevSelectedClass: string | null) => prevSelectedClass === classId ? null : classId);
+  };
+
   return (
     <table className={css.classesTable}>
       <thead>
         <tr><th>Class Name</th><th>Class Word</th><th></th></tr>
       </thead>
       <tbody>
+        {
+          classesData.map((classInfo) => {
+            const active = selectedClassId === classInfo.id;
+            return (
+              <>
+                <tr key={classInfo.id} className={clsx({ [css.activeRow]: active })}>
```

Comment:
I think this key needs to be hoisted into the parent fragment.

Reply:
Thanks, fixed in recent commits.